pkg/vpath/helpers_test.go

```diff
@@ -0,0 +1,34 @@
+package vpath
+
+import (
+	"testing"
+)
+
+func TestPathWalker(t *testing.T) {
+	type testCase struct {
+		Path  string
+		Parts []string
+	}
+
+	cases := []testCase{
+		{Path: "", Parts: nil},
+		{Path: "u1", Parts: []string{"u1"}},
+		{Path: "u1/u2", Parts: []string{"u1", "u2"}},
+		{Path: "u1/u2/u3", Parts: []string{"u1", "u2", "u3"}},
+	}
+
+	for _, test := range cases {
+		path := newPathWalker(test.Path)
+
+		for !path.Empty() {
+			if exp, got := test.Parts[0], path.Next(); exp != got {
+				t.Fatal("exp:", exp, "got:", got)
+			}
```

Comment:
Wouldn't `assert.Equal(t, test.Parts[0], path.Next())` do the same? It also won't stop the test here and will allow the rest of the test cases to execute.

Reply:
Sure, though, I'd probably want to break out of this inner `for` loop.